In [ ]:
# Loading Data:
import pandas as pd

try: # Running Locally, yakinkan "file_" berada di folder "data"
    dfJ = pd.read_csv("data/jakarta.csv", low_memory = False, encoding='utf8')
    dfD = pd.read_csv("data/depok.csv", low_memory = False, encoding='utf8')
    dfT = pd.read_csv("data/tangerang.csv", low_memory = False, encoding='utf8')
except: # Running in Google Colab
    !mkdir data
    !wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/jakarta.csv
    !wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/depok.csv
    !wget -P data/ https://raw.githubusercontent.com/taufiksutanto/EDA-Case-01-PreProcessing/master/data/tangerang.csv
    dfJ = pd.read_csv("data/jakarta.csv", low_memory = False, encoding='utf8')
    dfD = pd.read_csv("data/depok.csv", low_memory = False, encoding='utf8')
    dfT = pd.read_csv("data/tangerang.csv", low_memory = False, encoding='utf8')
    
dfJ.shape, dfD.shape, dfT.shape

((1024, 26), (998, 26), (128, 26))

In [ ]:
import pandas as pd
data_A = pd.read_csv('/content/data/depok.csv')
print(data_A.shape)
data_A

(998, 26)


created_at   LT     LB   KT   KM  garasi  carport     lokasi sertifikat  \
0    6-12-2021  165  200.0  4.0  2.0     NaN      1.0  Sukmajaya        SHM   
1    6-12-2021   50   45.0  2.0  1.0     NaN      NaN   Cipayung        SHM   
2    6-12-2021   50   36.0  2.0  1.0     NaN      1.0   Sawangan        SHM   
3    6-12-2021   50   45.0  2.0  1.0     NaN      NaN  Cimanggis        SHM   
4    6-12-2021   72   45.0  2.0  1.0     NaN      1.0   Sawangan        SHM   
..         ...  ...    ...  ...  ...     ...      ...        ...        ...   
993        NaN  NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
994        NaN  NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
995        NaN  NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
996        NaN  NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
997        NaN  NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   

     listrik  ... Unnamed: 16  Unnamed: 17 Unnamed: 18 Unnamed: 19  \
0     2200.0  ...         NaN          NaN         NaN         NaN   
1     1300.0  ...         NaN          NaN         NaN         NaN   
2     1300.0  ...         NaN          NaN         NaN         NaN   
3     1300.0  ...         NaN          NaN         NaN         NaN   
4     1300.0  ...         NaN          NaN         NaN         NaN   
..       ...  ...         ...          ...         ...         ...   
993      NaN  ...         NaN          NaN         NaN         NaN   
994      NaN  ...         NaN          NaN         NaN         NaN   
995      NaN  ...         NaN          NaN         NaN         NaN   
996      NaN  ...         NaN          NaN         NaN         NaN   
997      NaN  ...         NaN          NaN         NaN         NaN   

     Unnamed: 20  Unnamed: 21  Unnamed: 22  Unnamed: 23  Unnamed: 24  \
0            NaN          NaN          NaN          NaN          NaN   
1            NaN          NaN          NaN          NaN          NaN   
2            NaN          NaN          NaN          NaN          NaN   
3            NaN          NaN          NaN          NaN          NaN   
4            NaN          NaN          NaN          NaN          NaN   
..           ...          ...          ...          ...          ...   
993          NaN          NaN          NaN          NaN          NaN   
994          NaN          NaN          NaN          NaN          NaN   
995          NaN          NaN          NaN          NaN          NaN   
996          NaN          NaN          NaN          NaN          NaN   
997          NaN          NaN          NaN          NaN          NaN   

     Unnamed: 25  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN  
..           ...  
993          NaN  
994          NaN  
995          NaN  
996          NaN  
997          NaN  

[998 rows x 26 columns]

In [ ]:
data_A.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 998 entries, 0 to 997
Data columns (total 26 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   created_at   119 non-null    object 
 1   LT           120 non-null    object 
 2   LB           118 non-null    float64
 3   KT           118 non-null    float64
 4   KM           118 non-null    float64
 5   garasi       25 non-null     float64
 6   carport      94 non-null     float64
 7   lokasi       118 non-null    object 
 8   sertifikat   118 non-null    object 
 9   listrik      103 non-null    float64
 10  hadap        62 non-null     object 
 11  harga        118 non-null    float64
 12  URL          118 non-null    object 
 13  deskripsi    118 non-null    object 
 14  Unnamed: 14  0 non-null      float64
 15  Unnamed: 15  0 non-null      float64
 16  Unnamed: 16  0 non-null      float64
 17  Unnamed: 17  0 non-null      float64
 18  Unnamed: 18  0 non-null      float64
 19  Unnamed:

**berdasarkan informasi data, kualitas data dari daerah depok rendah karena banyak data kosong, maka untuk itu dara haruslah diperbaharui agar tidak memiliki data kosong**

In [ ]:
data_A.set_index('created_at')

LT     LB   KT   KM  garasi  carport     lokasi sertifikat  \
created_at                                                                
6-12-2021   165  200.0  4.0  2.0     NaN      1.0  Sukmajaya        SHM   
6-12-2021    50   45.0  2.0  1.0     NaN      NaN   Cipayung        SHM   
6-12-2021    50   36.0  2.0  1.0     NaN      1.0   Sawangan        SHM   
6-12-2021    50   45.0  2.0  1.0     NaN      NaN  Cimanggis        SHM   
6-12-2021    72   45.0  2.0  1.0     NaN      1.0   Sawangan        SHM   
...         ...    ...  ...  ...     ...      ...        ...        ...   
NaN         NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
NaN         NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
NaN         NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
NaN         NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   
NaN         NaN    NaN  NaN  NaN     NaN      NaN        NaN        NaN   

            listrik    hadap  ...  Unnamed: 16 Unnamed: 17 Unnamed: 18  \
created_at                    ...                                        
6-12-2021    2200.0  Selatan  ...          NaN         NaN         NaN   
6-12-2021    1300.0      NaN  ...          NaN         NaN         NaN   
6-12-2021    1300.0    Barat  ...          NaN         NaN         NaN   
6-12-2021    1300.0      NaN  ...          NaN         NaN         NaN   
6-12-2021    1300.0    Timur  ...          NaN         NaN         NaN   
...             ...      ...  ...          ...         ...         ...   
NaN             NaN      NaN  ...          NaN         NaN         NaN   
NaN             NaN      NaN  ...          NaN         NaN         NaN   
NaN             NaN      NaN  ...          NaN         NaN         NaN   
NaN             NaN      NaN  ...          NaN         NaN         NaN   
NaN             NaN      NaN  ...          NaN         NaN         NaN   

            Unnamed: 19  Unnamed: 20  Unnamed: 21  Unnamed: 22  Unnamed: 23  \
created_at                                                                    
6-12-2021           NaN          NaN          NaN          NaN          NaN   
6-12-2021           NaN          NaN          NaN          NaN          NaN   
6-12-2021           NaN          NaN          NaN          NaN          NaN   
6-12-2021           NaN          NaN          NaN          NaN          NaN   
6-12-2021           NaN          NaN          NaN          NaN          NaN   
...                 ...          ...          ...          ...          ...   
NaN                 NaN          NaN          NaN          NaN          NaN   
NaN                 NaN          NaN          NaN          NaN          NaN   
NaN                 NaN          NaN          NaN          NaN          NaN   
NaN                 NaN          NaN          NaN          NaN          NaN   
NaN                 NaN          NaN          NaN          NaN          NaN   

            Unnamed: 24  Unnamed: 25  
created_at                            
6-12-2021           NaN          NaN  
6-12-2021           NaN          NaN  
6-12-2021           NaN          NaN  
6-12-2021           NaN          NaN  
6-12-2021           NaN          NaN  
...                 ...          ...  
NaN                 NaN          NaN  
NaN                 NaN          NaN  
NaN                 NaN          NaN  
NaN                 NaN          NaN  
NaN                 NaN          NaN  

[998 rows x 25 columns]



*   Menghapus data duplikat



In [ ]:
data_A.duplicated().sum()

877

In [ ]:
data_A.drop_duplicates(inplace=True)
print(data_A.duplicated().sum())
print(data_A.shape)

0
(121, 26)


**dari data tersebut tersisa hanya 121 baris dan 26 kolom**



*   Menghapus Variable tidak penting



In [ ]:
data_A.drop("Unnamed: 15", axis=1, inplace=True)
data_A.drop("Unnamed: 19", axis=1, inplace=True)
data_A.drop("Unnamed: 20", axis=1, inplace=True)
data_A.drop("Unnamed: 21", axis=1, inplace=True)
data_A.drop("Unnamed: 22", axis=1, inplace=True)
data_A.drop("Unnamed: 23", axis=1, inplace=True)
data_A.drop("Unnamed: 24", axis=1, inplace=True)
data_A.drop("Unnamed: 25", axis=1, inplace=True)

In [ ]:
data_A

created_at                                               LT     LB   KT  \
0    6-12-2021                                              165  200.0  4.0   
1    6-12-2021                                               50   45.0  2.0   
2    6-12-2021                                               50   36.0  2.0   
3    6-12-2021                                               50   45.0  2.0   
4    6-12-2021                                               72   45.0  2.0   
..         ...                                              ...    ...  ...   
116  4-12-2021                                               50   50.0  2.0   
117  4-12-2021                                              100   15.0  3.0   
118        NaN                                              NaN    NaN  NaN   
120      NOTE:   - lokasi yang kami gunakan hanya per kecamatan    NaN  NaN   
121        NaN  -harga yang kami gunakan adalah per satuan juta    NaN  NaN   

      KM  garasi  carport     lokasi sertifikat  listrik    hadap   harga  \
0    2.0     NaN      1.0  Sukmajaya        SHM   2200.0  Selatan  2100.0   
1    1.0     NaN      NaN   Cipayung        SHM   1300.0      NaN   250.0   
2    1.0     NaN      1.0   Sawangan        SHM   1300.0    Barat   250.0   
3    1.0     NaN      NaN  Cimanggis        SHM   1300.0      NaN   250.0   
4    1.0     NaN      1.0   Sawangan        SHM   1300.0    Timur   350.0   
..   ...     ...      ...        ...        ...      ...      ...     ...   
116  1.0     NaN      NaN   Cipayung        SHM   1300.0  Selatan   185.0   
117  3.0     1.0      1.0     Cinere        SHM   3500.0      NaN  1750.0   
118  NaN     NaN      NaN        NaN        NaN      NaN      NaN     NaN   
120  NaN     NaN      NaN        NaN        NaN      NaN      NaN     NaN   
121  NaN     NaN      NaN        NaN        NaN      NaN      NaN     NaN   

                                                   URL  \
0    https://www.rumah123.com/properti/depok/hos928...   
1    https://www.rumah123.com/properti/depok/hos928...   
2    https://www.rumah123.com/properti/depok/hos928...   
3    https://www.rumah123.com/properti/depok/hos928...   
4    https://www.rumah123.com/properti/depok/hos928...   
..                                                 ...   
116  https://www.rumah123.com/properti/depok/hos927...   
117  https://www.rumah123.com/properti/depok/hos928...   
118                                                NaN   
120                                                NaN   
121                                                NaN   

                                             deskripsi  Unnamed: 14  \
0                        Rumah siap huni komplek Pelni          NaN   
1                  rumah minimalis termurah di citayam          NaN   
2              CLUSTER TERMURAH DI SAWANGAN KOTA DEPOK          NaN   
3     RUMAH SIAP HUNI HARGA NEGO DEKAT STASIUN CITAYAM          NaN   
4                        Rumah Cluster Siap Huni Depok          NaN   
..                                                 ...          ...   
116  rumah semi minimalis dua kamar siap huni di ci...          NaN   
117                 Dijual cepat rumah bagus siap huni          NaN   
118                                                NaN          NaN   
120                                                NaN          NaN   
121                                                NaN          NaN   

     Unnamed: 16  Unnamed: 17  Unnamed: 18  
0            NaN          NaN          NaN  
1            NaN          NaN          NaN  
2            NaN          NaN          NaN  
3            NaN          NaN          NaN  
4            NaN          NaN          NaN  
..           ...          ...          ...  
116          NaN          NaN          NaN  
117          NaN          NaN          NaN  
118          NaN          NaN          NaN  
120          NaN          NaN          NaN  
121          NaN          NaN          NaN  

[121 rows x 18 columns]

In [ ]:
data_A.isnull().sum()

created_at       2
LT               1
LB               3
KT               3
KM               3
garasi          96
carport         27
lokasi           3
sertifikat       3
listrik         18
hadap           59
harga            3
URL              3
deskripsi        3
Unnamed: 14    121
Unnamed: 16    121
Unnamed: 17    121
Unnamed: 18    121
dtype: int64

In [ ]:
data_A.describe()

LB          KT          KM     garasi    carport       listrik  \
count  118.000000  118.000000  118.000000  25.000000  94.000000    103.000000   
mean    92.618644    2.737288    1.940678   1.760000   1.670213   2147.572816   
std     92.630404    1.024720    0.980950   1.392839   1.061428   1166.261386   
min     15.000000    2.000000    1.000000   1.000000   1.000000    900.000000   
25%     46.000000    2.000000    1.000000   1.000000   1.000000   1300.000000   
50%     61.000000    3.000000    2.000000   1.000000   1.000000   2200.000000   
75%    102.250000    3.000000    2.000000   2.000000   2.000000   2200.000000   
max    600.000000    7.000000    5.000000   6.000000   7.000000  10000.000000   

             harga  Unnamed: 14  Unnamed: 16  Unnamed: 17  Unnamed: 18  
count   118.000000          0.0          0.0          0.0          0.0  
mean   1194.440678          NaN          NaN          NaN          NaN  
std    1474.387409          NaN          NaN          NaN          NaN  
min      41.000000          NaN          NaN          NaN          NaN  
25%     485.000000          NaN          NaN          NaN          NaN  
50%     810.000000          NaN          NaN          NaN          NaN  
75%    1137.500000          NaN          NaN          NaN          NaN  
max    9000.000000          NaN          NaN          NaN          NaN

**interpretasi:**
**berdasarkan data rumah di daerah depok yang paling banyak di jual adalah rumah dengan  **